multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ExtractBatchedServicesWithResolvedDynamicParametersStep.java

```diff
@@ -0,0 +1,130 @@
+                                                                                                                                      servicesCalculatedForDeployment);
+
+        List<CloudServiceInstanceExtended> resolvedServiceInstancesWithDynamicParameters = servicesCalculatedForDeployment.stream()
+                                                                                                                          .map(service -> resolveDynamicParametersOfServiceInstance(service,
+                                                                                                                                                                                    dynamicParametersWithResolvedExistingInstances))
+                                                                                                                          .collect(Collectors.toList());
+        setServicesToCreate(context, resolvedServiceInstancesWithDynamicParameters);
+
+        context.setVariable(Variables.DYNAMIC_RESOLVABLE_PARAMETERS, dynamicParametersWithResolvedExistingInstances);
+
+        return StepPhase.DONE;
+    }
+
+    @Override
+    protected String getStepErrorMessage(ProcessContext context) {
+        return Messages.ERROR_PREPARING_RESOURCES_FOR_PROCESSING_AND_RESOLVE_DYNAMIC_PARAMETERS;
+    }
+
+    private List<CloudServiceInstanceExtended> getServicesCalculatedForDeployment(ProcessContext context) {
+        List<Resource> batchToProcess = context.getVariable(Variables.BATCH_TO_PROCESS);
+        return getServicesCloudModelBuilder(context).build(batchToProcess);
+    }
+
+    private ServicesCloudModelBuilder getServicesCloudModelBuilder(ProcessContext context) {
+        CloudHandlerFactory handlerFactory = StepsUtil.getHandlerFactory(context.getExecution());
+        DeploymentDescriptor deploymentDescriptor = context.getVariable(Variables.COMPLETE_DEPLOYMENT_DESCRIPTOR);
+        String namespace = context.getVariable(Variables.MTA_NAMESPACE);
+
+        return handlerFactory.getServicesCloudModelBuilder(deploymentDescriptor, namespace);
+    }
+
+    private Set<DynamicResolvableParameter>
+            resolveDynamicPramatersWithExistingInstances(CloudControllerClient client,
+                                                         Set<DynamicResolvableParameter> dynamicResolvableParameters,
+                                                         List<CloudServiceInstanceExtended> servicesCalculatedForDeployment) {
+        Map<String, String> existingServiceGuids = getExistingServiceGuidsIfNeeded(client, dynamicResolvableParameters,
+                                                                                   servicesCalculatedForDeployment);
+        Set<DynamicResolvableParameter> dynamicParametersWithResolvedExistingInstances = new HashSet<>(dynamicResolvableParameters);
```

Comment:
similar comment - maybe 'resolvedDynamicParameters' will be a more readable variable name for method called 'resolveDynamicPramatersWithExistingInstances'